src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -482,6 +482,20 @@ bool TensorIterator::needPrepareParams() const {
             return true;
     }
 
+    // If input body shape is changed after execution, it means that the execution influences shapes.
+    // Thus, for these cases we should always reset this input shape after execution even if Loop/TI has the same input shapes,
+    // the same input data to have correct input shape for the next InferRequest (execution)
+    // For example, it's a valid case when body has explicit concatenation Result:
+    // Parameter0     ...
+    //    \         /
+    //       Concat
+    //         |
+    //       Result
+    //  (with back edge to Parameter0)
+    if (body_inshapes_reset) {
```

Comment:
Not sure I understand why should we compute this condition on execution stage of previous infer_request. It seems to be unclear solution.
Cannot we just check input and body shapes equality here and return false if they are different?

Reply:
Please put this as a comment into the code